File: example-module-5/exam/src/app/model/car.ts
import {HomeCar} from "./home-car";

export interface Car {
  id?:number;
  loaiXe?:string;
  diaDiemDi?:string;
  diaDiemDen?:string;
  gioKhoiHanh?:string;
  gioDen?:string;
  nhaXe?:HomeCar;
}

File: conect_backend_module5/src/main/java/com/example/model/Car.java
package com.example.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Car {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String loaiXe;
    private String diaDiemDi;
    private String diaDiemDen;
    private String gioKhoiHanh;
    private String gioDen;


    @ManyToOne
    @JoinColumn(name = "homeCar",referencedColumnName = "id")
    private HomeCar nhaXe;
}
